kube-client/src/client/auth/mod.rs

```diff
@@ -474,22 +474,32 @@ fn token_from_gcp_provider(provider: &AuthProviderConfig) -> Result<ProviderToke
     }
 }
 
-fn extract_value(json: &serde_json::Value, path: &str) -> Result<String, Error> {
-    let pure_path = path.trim_matches(|c| c == '"' || c == '{' || c == '}');
-    match jsonpath_select(json, &format!("${pure_path}")) {
-        Ok(v) if !v.is_empty() => {
-            if let serde_json::Value::String(res) = v[0] {
-                Ok(res.clone())
-            } else {
-                Err(Error::AuthExec(format!(
-                    "Target value at {pure_path:} is not a string"
-                )))
-            }
-        }
-
-        Err(e) => Err(Error::AuthExec(format!("Could not extract JSON value: {e:}"))),
+fn extract_value(json: &serde_json::Value, context: &str, path: &str) -> Result<String, Error> {
+    let parsed_path = path
+        .trim_matches(|c| c == '"' || c == '{' || c == '}')
+        .parse::<JsonPathInst>()
+        .map_err(|err| {
+            Error::AuthExec(format!(
+                "Failed to parse {context:?} as a JsonPath: {path}\n
+                 Error: {err}"
+            ))
+        })?;
+
+    let res = parsed_path.find_slice(json);
+
+    let Some(res) = res.into_iter().next() else {
+        return Err(Error::AuthExec(format!(
+            "Target {context:?} value {path:?} not found"
+        )));
+    };
 
-        _ => Err(Error::AuthExec(format!("Target value {pure_path:} not found"))),
+    if let Some(val) = res.as_str() {
+        Ok(val.to_owned())
+    } else {
+        Err(Error::AuthExec(format!(
+            "Target {:?} value {:?} is not a string: {:?}",
```

Comment:
```suggestion
            "Target {context:?} value {path:?} is not a string: {res:?}",
```

Reply:
Yes, `res` needs to be dereferenced.
I think, it is confusing when inline symbols are mixed with expressions that are passed as arguments.
Even clippy only suggests inlining only if everything can be inlined.
Controlled by [`allow-mixed-uninlined-format-args`](https://doc.rust-lang.org/nightly/clippy/lint_configuration.html#allow-mixed-uninlined-format-args).

But, I can make this change if this is the style this project uses.

Reply:
Fair point. Personally I think it helps with readability, especially since it is only the last argument.

I looked for a precedent in the code base and couldn't find one (only a bunch of old stuff which now could be inlined).

So I am fine with it either way.